example-auth/README.md

```diff
@@ -3,395 +3,497 @@ This example is to showcase [Appwrite's JS API](https://github.com/appwrite/sdk-
+
+```ts
+import { Client } from "appwrite"
+
+const client = new Client()
+  .setEndpoint(import.meta.env.VITE_APPWRITE_URL)
+  .setProject(import.meta.env.VITE_APPWRITE_PROJECT);
 ```
-A deeper inspection of this code can be found in the comments within it, 
-
-TL:DR: Create a appwrite SDK Instance and initalise it with the endpoint and ProjectID of the project we are working with then export this for usage outside of the file.
-
-## Creating the App.js
-We are now going to replace the `src/App.js` with our own, doing so we will turn the object from a function react component to a class based one aswell as adding a bunch of logic to the app which will be used later by components.
-
-```js
-import React from 'react';
-import { appwrite } from './utils';
-import './App.css';
-
-import { Login } from './components/Login';
-import { Profile } from './components/Profile';
-
-class App extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = { // Create the variables we will use later.
-            userprofile: false,
-            error: false
-        };
-    };
-
-    // Get userdata function.
-    async getUserdata() {
-        try {
-            const response = await appwrite.account.get(); // Request to appwrite server to see if we are logged in.
-            this.setState({ userprofile: response }); // If Logged in then set the returned profile to the userprofile variable in state.
-        } catch (err) { // If we are not logged in or another error occoured then catch(err)
-            if (err.toString() === 'Error: Unauthorized') return; // If not logged in then do nothing.
-            this.setState({ error: err.toString() }); // If it's another error then set the error variable in state.
-            console.error(err); // and also console.error the error for clearer debugging.
-        }
-    }
 
-    // Login function
-    async login(email, password) {
-        try {
-            // Set error to false so if we are successful the error doesn't perist making bad UX Design.
-            // also set the loading prop to true to signal to the user we are processing his request.
-            await this.setState({ error: false })
-
-            // Create the session, if this fails it will error and be caught by the catch(err).
-            await appwrite.account.createSession(
-                email,
-                password
-            );
-            // If all is successful then get the userdata.
-            this.getUserdata();
-        } catch (err) {
-            await this.setState({ error: 'Invalid Credentials' }) // If login fails then show user the login was not successful.
-            console.error(err) // also console error for debugging purposes.
-        }
-    }
+In this codeblock we are using Vite's approach for environment variables. To do this, we're going to create a `.env` file at the root of our project and then populate it with the following information:
```

Comment:
It might be a great idea to create a file called `.env-template` in the repo to show what the user is expected to create.

For example something like
```
VITE_APPWRITE_URL=https://localhost/v1
VITE_APPWRITE_PROJECT=demos-for-react
```

Reply:
<img width="230" alt="image" src="https://user-images.githubusercontent.com/3102249/195365029-f5c868dd-0d80-4b0a-b24a-e7594cbf06e0.png">

Reply:
I created the template file and added a note in the README